test/src/test/java/org/corfudb/runtime/view/RuntimeViewsTest.java

```diff
@@ -0,0 +1,63 @@
+package org.corfudb.runtime.view;
+
+
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.MultiCheckpointWriter;
+import org.corfudb.runtime.collections.CorfuTable;
+import org.corfudb.runtime.object.CorfuCompileProxy;
+import org.corfudb.runtime.object.ISMRStream;
+import org.corfudb.runtime.object.StreamViewSMRAdapter;
+import org.corfudb.runtime.object.VersionLockedObject;
+import org.corfudb.runtime.view.stream.BackpointerStreamView;
+import org.ehcache.sizeof.SizeOf;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Created by maithem on 11/16/18.
+ */
+
+public class RuntimeViewsTest extends AbstractViewTest {
+
+    @Test
+    public void testRuntimeGC() throws Exception {
+
+        CorfuRuntime rt = getDefaultRuntime();
+
+        CorfuTable<String, String> table = rt.getObjectsView()
+                .build()
+                .setType(CorfuTable.class)
+                .setStreamName("table1")
+                .open();
+
+        SizeOf sizeOf = SizeOf.newInstance();
+        long sizeAfterCreation = sizeOf.deepSizeOf(table);
+        rt.runRuntimeGC();
+        assertThat(sizeAfterCreation).isLessThanOrEqualTo(sizeOf.deepSizeOf(table));
+
+        final int numWrites = 100;
+
+        for (int x = 0; x < numWrites; x++) {
+            table.put(String.valueOf(x), String.valueOf(x));
+        }
+
+        long sizeOfTableAfterWrite = sizeOf.deepSizeOf(table);
+
+        MultiCheckpointWriter mcw = new MultiCheckpointWriter();
+        mcw.addMap(table);
+        Token trimMark = mcw.appendCheckpoints(rt, "cp1");
+        rt.getAddressSpaceView().prefixTrim(trimMark.getSequence());
+        rt.getParameters().setRuntimeGCPeriod(Duration.ofMinutes(0));
+        rt.runRuntimeGC();
+        long sizeOfTableAfterGc = sizeOf.deepSizeOf(table);
+
+        assertThat(sizeOfTableAfterGc).isLessThan(sizeOfTableAfterWrite);
+        assertThat(rt.getAddressSpaceView().getReadCache().asMap()).isEmpty();
```

Comment:
Can we also assert on the resolvedQueue being trimmed in the StreamsView?

Reply:
Exposing the context to the test is very messy. Line 60 tests this indirectly. I made the condition tighter:
assertThat(sizeOfTableAfterGc).isLessThan(sizeOfTableAfterWrite - (numWrites * Long.BYTES));